insights/parsers/satellite_settings.py

```diff
@@ -0,0 +1,77 @@
+``psql -d foreman -c 'select name, value, "default" from settings where name in (\'destroy_vm_on_host_delete\', \'unregister_delete_host\')'``.
+
+Sample output of ``psql -d foreman -c 'select name, value, "default" from settings where name in (\'destroy_vm_on_host_delete\', \'unregister_delete_host\')'``::
+
+            name            |  value   |  default
+    ---------------------------+----------+-----------
+    unregister_delete_host    | --- true+| --- false+
+                            | ...      | ...
+    destroy_vm_on_host_delete |          | --- true +
+                            |          | ...
+    (2 rows)
+
+
+Examples::
+
+    >>> type(settings)
+    <class 'insights.parsers.satellite_settings.SatelliteSettings'>
+    >>> 'unregister_delete_host' in settings
+    True
+    >>> settings['unregister_delete_host']
+    True
+    >>> len(settings)
+    2
+    >>> settings['destroy_vm_on_host_delete']
+    True
+"""
+
+import os
+import yaml
+
+from insights import parser, CommandParser
+from insights.specs import Specs
+from insights.parsers import SkipException, ParseException
+
+
+@parser(Specs.satellite_settings)
+class SatelliteSettings(CommandParser, dict):
+    """
+    Read the ``psql -d foreman -c '"select name, value, "defualt" from settings where name in ('destroy_vm_on_host_delete', 'unregister_delete_host')"``
+    and save the data to itself as a dict.
+    """
+
+    def parse_content(self, content):
+        columns = ['name', 'value', 'default']
+        if len(content) >= 4:
+            columns_in_data = [item.strip() for item in content[0].split('|')]
+            if columns_in_data == columns:
+                rows = content[2:]  # skip title and '---'
+                name = val = new_val = ''
+                for row in rows:
+                    if '|' not in row:
+                        continue
+                    data = row.split('|')
+                    if len(data) != len(columns):
+                        raise SkipException("The returned data is in bad format")
+                    if not name:
+                        name, val, default_val = [item.strip() for item in data]
+                    else:
+                        _, new_val, new_default_val = [item.strip() for item in data]
+                        if val.endswith('+'):
+                            val = val.rstrip(' +') + os.linesep + new_val
+                        if default_val.endswith('+'):
+                            default_val = default_val.rstrip(' +') + os.linesep + new_default_val
+                    if not (val.endswith('+') or default_val.endswith('+')):
+                        real_val = val or default_val
+                        try:
+                            self[name] = yaml.safe_load(real_val)
+                        except Exception:
+                            raise ParseException('Could not parse the value for satellite settings of name %s ' % name)
+                        name = val = new_val = ''
```

Comment:
@huali027 this seems more complicated that it needs to be.  You know what columns are being selected and the order, and you know the row values you want to collect so why not just look for the rows with the 2 keys `[destroy_vm_on_host_delete', 'unregister_delete_host']` and then check for the 'true` and `false` values.  It just seems like it could be simplified given that you know what data is being collected.  What do you think?

Reply:
@bfahr, let me explain more:
1. first I check the columns are in the same order with what expected. Because if customers didn't change the setting, the default column should be used, if changes are done, then the value column should be used.
2. currently,  I only need these two settings, so I limit the SQL to it. If in the parser code, I hard code these two column, and consider the value is just boolean and later if any new setting is needed, we have to update the parser too. For these two settings, the value is just a boolean value, but for other settings, the value may be string or int or other types. But all of them are in the yaml format. 
For example:
```
               name                | value |                              default                              
-----------------------------------+-------+-------------------------------------------------------------------
 foreman_tasks_proxy_batch_size    |       | --- 100                                                          +
                                   |       | ...
 foreman_tasks_troubleshooting_url |       | --- https://access.redhat.com/solutions/satellite6-tasks#%{label}+
                                   |       | ...
(2 rows)

```
Do u think we should only consider the two settings now without considering more?

Reply:
There is always a trade-off between doing what you need today versus trying to anticipate future possibilities.  If you think its likely that you'll add more columns in the future, and you can write a parser that implements those future data types, then it makes sense to have more complicated parsing.  If you go with the more complex parser please make sure you include sufficient testing to ensure it fails if something is added you don't anticipate.

Reply:
@bfahr, sure, I'll add more tests too.